### ISSUE TITLE
Top-level errors `path` field has incorrect value in case of nested service call

### ISSUE THREAD

**Issue by mpospelov** [apollographql/federation#354]
* Package name: **apollo-gateway** occurs on `master` and latest versions
* Here a commit with failing test:
  - https://github.com/mpospelov/apollo-server/commit/cd9cc2424026ab690a72d8874054747928d489b6
  - we are fetching the following query
  ```gql
  query {
    topReviews(first: 10) {
      author {
        name
      }
    }
  }
  ```
  - one of authors has error while fetching `name` field in `accounts` schema
* The expected behavior: `expect(response).toHaveProperty('errors.0.path', ['topReviews', 7, 'author', 'name'])`
* The actual behavior:  `["_entities", 7, "name"]`
* Runnable reproduction: 
  1. clone my branch https://github.com/mpospelov/apollo-server/tree/nested-services-path-building-issue
  2. run `cd packages/apollo-gateway && npx jest src/__tests__/executeQueryPlan.test.ts`
* Reasoning behind this behaviour:
  1. In our project, there are multiple services which receive requests via apollo-gateway
  2. Each service can raise authorization field error based on current user permission
  3. In order to show errors properly on front-end, `errors.*.path` field should comply with graphql specs https://spec.graphql.org/June2018/#sec-Errors

**Comment by jonahmolinski** [comment]
Is there any word on this? Or a work around anyone has come up with? Cheers!

**Comment by prowe** [comment]
My first look at the code, but it seems something could be done in this method to map the error paths: https://github.com/apollographql/federation/blob/813101200012f9268fcafb65eeaedf27ca7479b7/gateway-js/src/executeQueryPlan.ts#L192

**Comment by benweatherman** [comment]
Howdy to everyone following this issue! We're planning on rolling this into the `2.2.0` release, which is still in the planning phase. We don't know the specific implementation path yet (the linked PR #988 seems like a great start), but we'll start making more progress here within the next few weeks.

**Comment by EXPEylazzari** [comment]
Hi @benweatherman !

Do you have an ETA to share when the `2.2.0` release will be made available? On the [milestones](https://github.com/apollographql/federation/milestones) page of the project, there is no due date set for that release. I see the `2.1.0` release is just around the corner, due 2 days from now on July 8th, but hoping that the `2.2.0` release won't be too far off so that we can benefit from this fix at some point in the near future. 

Thank you very much!

**Comment by benweatherman** [comment]
Howdy @EXPEylazzari! No ETA yet on _when_ this will release, but it's the next feature we're working on. Once it's finished, it'll go into whatever the next release is.

For more context, our main focus for `2.1.0` is composing user directives #1893, which has changed several times during  implementation. That's causing the `2.1.0` release to take longer than we originally anticipated. So it's possible this issue will be released in `2.1.0`, depending on when #1893 wraps up.

**TL;DR** we're working on this next and it'll be released in whatever version follows its completion.

**Comment by korinne** [comment]
Hi all! We recently fixed this issue in https://github.com/apollographql/federation/pull/2304, which has been delivered in [Federation v2.3](https://github.com/apollographql/federation/blob/main/gateway-js/CHANGELOG.md#230-beta3). With that, I'll be closing this issue, but feel free to open any new issues if your use cases are not fully met. Thanks!
